pkg/labels/matcher.go

```diff
@@ -74,6 +74,9 @@ func NewMatcher(t MatchType, n, v string) (*Matcher, error) {
 }
 
 func (m *Matcher) String() string {
+	if !model.LabelName(m.Name).IsValid() {
```

Comment:
I'm not sure I understand what this does, would it be possible to explain it in a comment?

Reply:
Sure! The idea is to check whether the name is Prometheus-compatible and return it without quotes (same as it does now), and wrap it in double quotes otherwise.